src/transformers/pipelines/base.py

```diff
@@ -278,7 +278,7 @@ def infer_framework_load_model(
         if isinstance(model, str):
             raise ValueError(f"Could not load model {model} with any of the following classes: {class_tuple}.")
 
-    framework = "tf" if "keras.engine.training.Model" in str(inspect.getmro(model.__class__)) else "pt"
+    framework = "tf" if "TFPreTrainedModel" in str(inspect.getmro(model.__class__)) else "pt"
```

Comment:
This code also assumed internal Keras namespaces wouldn't move around. I've used "TFPreTrainedModel" as a string that should be more constant, but we could also just use "tensorflow" or "keras" if there are cases where the models we want to use will not inherit from `TFPreTrainedModel`.

Reply:
@sgugger this is one bit I was uncertain about - does every model we could use in a pipeline inherit from `PreTrainedModel` or `TFPreTrainedModel`? I assume yes but it felt a little dangerous.

Reply:
Alright, let me try to figure out something that covers all the relevant versions without catching anything else.